pywbem_mock/_mainprovider.py

```diff
@@ -0,0 +1,4214 @@
+                _format("Class {0!A} in namespace {1!A} not found. "
+                        "Cannot delete instance {2!A}",
+                        iname.classname, namespace, iname))
+
+        if not instance_store.exists(iname):
+            raise CIMError(
+                CIM_ERR_NOT_FOUND,
+                _format("Instance {0!A} not found in CIM repository namespace "
+                        "{1!A}", iname, namespace))
+
+        # Handle namespace deletion, currently hard coded.
+        # Issue #2062 TODO/AM 8/18 Generalize the hard coded handling into
+        # provider concept
+        classname_lower = iname.classname.lower()
+        if classname_lower == 'pg_namespace':
+            ns_classname = 'PG_Namespace'
+        elif classname_lower == 'cim_namespace':
+            ns_classname = 'CIM_Namespace'
+        else:
+            ns_classname = None
+        if ns_classname:
+            namespace = iname.keybindings['Name']
+
+            # Reflect the namespace deletion in the CIM repository
+            # This will check the namespace for being empty.
+            self.remove_namespace(namespace)
+
+        # Delete the instance from the CIM repository
+        instance_store.delete(iname)
+
+    ####################################################################
+    #
+    #   EnumerateInstances server request
+    #
+    ####################################################################
+
+    def EnumerateInstances(self, namespace, ClassName, LocalOnly=None,
+                           DeepInheritance=None, IncludeQualifiers=None,
+                           IncludeClassOrigin=None, PropertyList=None):
+        # pylint: disable=line-too-long
+        """
+        Implements a WBEM server responder for
+        :meth:`~pywbem.WBEMConnection.EnumerateInstances`.
+
+        Gets a list of subclasses of `ClassName` or of the class hiearchy root
+        in the CIM repository then executes get instance for each to create
+        the list of instances to be returned.
+
+        Parameters:
+
+          namespace (:term:`string`):
+            Name of the namespace in which the CIM instance will be created
+            (case independent).
+
+            Leading and trailing slash characters will be stripped. The lexical
+            case will be preserved.
+
+          ClassName (:term:`string`):
+            Name of the class for which instances are to be enumerated (case
+            independent).
+
+          TODO: confirm this parameter.  In this case we always use False
+          LocalOnly (:class:`py:bool`):
+            Controls the exclusion of inherited properties from the returned
+            instances.
+
+            This parameter has been deprecated in :term:`DSP0200` and should be
+            set to `False` by the caller.  This responder ignores the value
+            of this parameter and acts as if the parameter is set `False`,
+            returning all properties include inherited properties
```

Comment:
Treating it as False is consistent with DSP0200 1.4, but not consistent with 1.0.

From DSP0200 1.4: "The DMTF strongly recommends that WBEM clients set LocalOnly to false and do not use this parameter to filter the set of properties returned. To minimize the impact of this recommendation on WBEM clients, a WBEM server may choose to treat the value of the LocalOnly parameter as false for all requests. A WBEM server shall consistently support a single interpretation of the LocalOnly parameter."

Reply:
Sorry if my comment was confusing. I just wanted to give the facts and I did not intend to argue in favor of implementing a default value of True. Also, I don't think we should repeat all of that.

So, bottom line:
* I think we should implement a default value of False consistently (that means for EnumerateInstances and EnumerateInstanceNames)
* I think we should have a short sentence explaining that the default of False is implemented in this server because DSP0200 1.4 recommends that this value should be used by clients.

Reply:
Expanded comment on the usage issues and defined global flag that always sets it to False

Reply:
ok